src/core/selinux-access.c

```diff
@@ -137,31 +137,30 @@ _printf_(2, 3) static int log_callback(int type, const char *fmt, ...) {
 }
 
 static int access_init(sd_bus_error *error) {
-
         if (!mac_selinux_use())
                 return 0;
 
         if (initialized)
                 return 1;
 
         if (avc_open(NULL, 0) != 0) {
-                int saved_errno = errno;
                 bool enforce;
 
-                enforce = security_getenforce() != 0;
-                log_full_errno(enforce ? LOG_ERR : LOG_WARNING, saved_errno, "Failed to open the SELinux AVC: %m");
+                {
+                        PROTECT_ERRNO;
+                        enforce = security_getenforce() != 0;
+                }
+
+                log_full_errno(enforce ? LOG_ERR : LOG_WARNING, errno, "Failed to open the SELinux AVC: %m");
```

Comment:
I think in other places in cases like this we just defined `int r` up top, and then did `r = -errno;` right after the `avc_open()` and then use `r` further down...

Reply:
I changed to code to use an `r` now. The code is about the same length, either way seems to work OK.